processors/filterprocessor/gen/generate.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/algorand/go-algorand/data/transactions"
+)
+
+// recursiveTagFields recursively gets all field names in a struct
+// Output will contain a key of the full tag along with the fully qualified struct
+func recursiveTagFields(theStruct interface{}, Output *map[string]string, tagLevel []string, fieldLevel []string) {
+	rStruct := reflect.TypeOf(theStruct)
+	numFields := rStruct.NumField()
+	for i := 0; i < numFields; i++ {
+		field := rStruct.Field(i)
+		name := field.Name
+
+		var tagValue string
+		var foundTag bool
+		// If there is a codec tag...
+		if tagValue, foundTag = field.Tag.Lookup("codec"); foundTag {
+
+			vals := strings.Split(tagValue, ",")
+			// Get the first value (the one we care about)
+			tagValue = vals[0]
+			// If it is empty ignore it
+			if tagValue == "" {
+				continue
+			}
+
+			fullTag := strings.Join(append(tagLevel, tagValue), ".")
+			(*Output)[fullTag] = strings.Join(append(fieldLevel, name), ".")
+		}
+
+		if field.Type.Kind() == reflect.Struct {
+			var passedTagLevel []string
+			if foundTag {
+				passedTagLevel = append(tagLevel, tagValue)
+			} else {
+				passedTagLevel = tagLevel
+			}
+			recursiveTagFields(reflect.New(field.Type).Elem().Interface(), Output, passedTagLevel, append(fieldLevel, name))
+		}
+	}
+}
+
+// usage:
+// go run generate.go packagename outputfile
+func main() {
+
+	var packageName string
+	var outputFilepath string
+
+	if len(os.Args) == 3 {
+		packageName = os.Args[1]
+		outputFilepath = os.Args[2]
+	}
+
+	if packageName == "" {
+		packageName = "NULL"
+	}
+
+	output := make(map[string]string)
+	tagLevel := []string{}
+	fieldLevel := []string{}
+
+	recursiveTagFields(transactions.SignedTxnInBlock{}, &output, tagLevel, fieldLevel)
+
+	var err error
+	var bb bytes.Buffer
+
+	initialStr := `// Code generated via go generate. DO NOT EDIT.
+
+package %s
+
+import "github.com/algorand/go-algorand/data/transactions"
+
+// SignedTxnMap generates a map with the key as the codec tag and the value as a function
+// that returns the associated variable
+var SignedTxnMap = map[string]func(*transactions.SignedTxnInBlock) interface{}{
```

Comment:
Could you create a benchmark to compare the performance of a switch statement vs the map of functions? From our conversation I was expecting something like
```go
switch tag {
case "txn.snd":
    return t.SignedTxn.Sender
case "txn.rcv":
...
```

The top 2 results for "golang map vs switch" seem to say opposite things:
https://hashrocket.com/blog/posts/switch-vs-map-which-is-the-better-way-to-branch-in-go
https://stackoverflow.com/questions/46789259/map-vs-switch-performance-in-go

I wonder if the extra overhead of a function call would change things.